private/pkg/transport/http/httpserver/httpserver.go

```diff
@@ -0,0 +1,180 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package httpserver
+
+import (
+	"context"
+	"crypto/tls"
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/go-chi/chi/v5"
+	"go.uber.org/zap"
+)
+
+const (
+	// DefaultShutdownTimeout is the default shutdown timeout.
+	DefaultShutdownTimeout = 10 * time.Second
+	// DefaultReadHeaderTimeout is the default read header timeout.
+	DefaultReadHeaderTimeout = 30 * time.Second
+	// DefaultIdleTimeout is the amount of time an HTTP/2 connection can be idle.
+	DefaultIdleTimeout = 3 * time.Minute
+)
+
+// Mapper initializes a Router.
+type Mapper interface {
+	Map(chi.Router) error
+}
+
+// MapperFunc is a function for a Mapper
+type MapperFunc func(chi.Router) error
+
+// Map implements Mapper.
+func (m MapperFunc) Map(router chi.Router) error {
+	return m(router)
+}
+
+// NewHTTPHandlerMapper returns a new Mapper for the http.Handler.
+func NewHTTPHandlerMapper(
+	handler http.Handler,
+	options ...HTTPHandlerMapperOption,
+) Mapper {
+	return newHTTPHandlerMapper(handler, options...)
+}
+
+// HTTPHandlerMapperOption is an option for a new HTTPHandlerMapper.
+type HTTPHandlerMapperOption func(*httpHandlerMapper)
+
+// HTTPHandlerMapperWithPrefix returns a new HTTPHandlerMapperOption that uses the prefix.
+func HTTPHandlerMapperWithPrefix(prefix string) HTTPHandlerMapperOption {
+	return func(httpHandlerMapper *httpHandlerMapper) {
+		httpHandlerMapper.prefix = prefix
+	}
+}
+
+// NewPrefixedHTTPHandlerMapper returns a new Mapper.
+func NewPrefixedHTTPHandlerMapper(
+	handlers []PrefixedHTTPHandler,
+	options ...PrefixedHTTPHandlerMapperOption,
+) Mapper {
+	return newPrefixedHTTPHandlerMapper(
+		handlers,
+		options...,
+	)
+}
+
+// PrefixedHTTPHandlerMapperOption is an option for a new PrefixedHTTPHandlerMapper.
+type PrefixedHTTPHandlerMapperOption func(*prefixedHTTPHandlerMapper)
+
+// PrefixedHTTPHandlerMapperWithGlobalPrefix returns a new PrefixedHTTPHandlerMapperOption that uses the global prefix.
+func PrefixedHTTPHandlerMapperWithGlobalPrefix(globalPrefix string) PrefixedHTTPHandlerMapperOption {
+	return func(prefixedHTTPHandlerMapper *prefixedHTTPHandlerMapper) {
+		prefixedHTTPHandlerMapper.globalPrefix = globalPrefix
+	}
+}
+
+// PrefixedHTTPHandler is an http.Handler with a path prefix.
+//
+// A router should route all requests with the path prefix to this handler.
+type PrefixedHTTPHandler interface {
+	http.Handler
+	PathPrefix() string
+}
+
+// NewPrefixedHTTPHandler creates a new PrefixedHTTPHandler using a handler and prefix.
+func NewPrefixedHTTPHandler(handler http.Handler, prefix string) PrefixedHTTPHandler {
+	return newPrefixedHTTPHandler(handler, prefix)
+}
```

Comment:
Some of these are unused in core, lets make sure we don't port that over here and just delete them instead.